sirepo/package_data/static/js/elegant.js

```diff
@@ -1443,3 +1443,75 @@ SIREPO.app.directive('srBunchEditor', function(appState, panelState) {
         },
     };
 });
+
+
+SIREPO.app.directive('viewLogIframe', function(appState, requestSender) {
+    return {
+        restrict: 'A',
+        scope: {},
+        template: `
+            <a href data-ng-click="viewLog()">View Log</a>
+            <div class="modal fade" id="sr-iframe-text-view" tabindex="-1" role="dialog">
+              <div class="modal-dialog modal-lg">
+                <div class="modal-content">
+                  <div class="modal-header bg-warning">
+                    <span class="lead modal-title text-info">Log</span>
+                    <div class="sr-panel-options pull-right">
+                      <a data-ng-href="{{ downloadLog() }}" target="_blank">
+                        <span class="sr-panel-heading glyphicon glyphicon-cloud-download"></span>
+                      </a>
+                      <button type="button" class="close" data-dismiss="modal" style="margin-left: 10px">
+                        <span>&times;</span>
+                      </button>
+                    </div>
+                  </div>
+                  <div class="modal-body" style="padding: 0">
+                    <iframe id="sr-text-iframe"
+                      style="border: 0; width: 100%; height: 80vh" src=""></iframe>
+                  </div>
+                </div>
+              </div>
+            </div>
+        `,
+        controller: function(appState, requestSender, $scope) {
+
+            function setIFrameHTML(html) {
+                $('#sr-text-iframe').contents().find('html').html(html);
+            }
+
+            function getDownloadURL(model, index) {
+                if (! appState.isLoaded() || ! model) {
+                    return '';
+                }
+                return requestSender.formatUrl('downloadDataFile', {
+                    '<simulation_id>': appState.models.simulation.simulationId,
+                    '<simulation_type>': SIREPO.APP_SCHEMA.simulationType,
+                    '<model>': model,
+                    '<frame>': index,
+                });
+            }
+
+            $scope.viewLog = function(model, filename) {
+                if (! appState.isLoaded()) {
+                    return '';
+                }
+                
+                requestSender.sendAnalysisJob(
+                    appState,
+                    (data) => {
+                        setIFrameHTML(data.html);
+                    },
+                    {
+                        method: 'log_to_html',
+                        computeModel: appState.models.simulationStatus.animation.computeModel,
```

Comment:
I think the more "correct" way to get this is to specify `elegantService` in the arguments to the controller function above. Then here to say `elegantService.computeModel()`. 

Every app has a `<app-name>Service` in Sirepo. And every one of those (or almost every one) has an abstraction for getting the computeModel.

Reply:
Got it good to know.